src/rebar_prv_eunit.erl

```diff
@@ -40,14 +40,18 @@ init(State) ->
 
 -spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
 do(State) ->
-    Tests = prepare_tests(State),
-    %% inject `eunit_first_files`, `eunit_compile_opts` and any
-    %% directories required by tests into the applications
-    NewState = inject_eunit_state(State, Tests),
-    case compile(NewState) of
-        %% successfully compiled apps
-        {ok, S} -> do(S, Tests);
-        Error   -> Error
+    try
+        Tests = prepare_tests(State),
+        %% inject `eunit_first_files`, `eunit_compile_opts` and any
+        %% directories required by tests into the applications
+        NewState = inject_eunit_state(State, Tests),
+        case compile(NewState) of
+            %% successfully compiled apps
+            {ok, S} -> do(S, Tests);
+            Error   -> Error
+        end
+    catch
+        throw:Reason -> {error, Reason}
```

Comment:
it would be clearer and easier to maintain in the long run if the `catch` of the exception was moved closer to where it's generated than the top level -- the `{error, Reason}` tuple format is already supported from a lower layer, and getting into that flow would probably be the simplest way to prevent mistakenly breaking the handling of this new error value.